### PR TITLE
Streamline discussions user support

### PR DESCRIPTION
**Pull request by GiudGiud**
refs #26980

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug-report.md
Comment (GiudGiud) [inline .github/ISSUE_TEMPLATE/bug-report.md:1]: I ll remove this change once new templates are in

**Review by permcody — APPROVED** [review]
I have no issue with this. I would recommend we get our framework lead (@lindsayad) involved too.

**Comment by milljm** [comment]
We've hit this requirement now have we? I am fine with a checklist. I think it will help.

**Comment by milljm** [comment]
Perhaps a direct link to our troubleshooting section would be prudent as well?

```pre
       - label: I have search the MOOSE website and [troubleshooting](https://mooseframework.inl.gov/help/troubleshooting.html) does not answer my question
          required: true
```

**Comment by GiudGiud** [comment]
Good idea. I ll add the other help pages

**Review by lindsayad — COMMENTED** [review]
I think this is fantastic. I clicked briefly into the links on the associated Slack thread so may have easily missed it, but is there a picture of how this renders when the user is filling it out?

**Comment by milljm** [comment]
> I think this is fantastic. I clicked briefly into the links on the associated Slack thread so may have easily missed it, but is there a picture of how this renders when the user is filling it out?

I am not sure... good question! Conda-Forge is doing it here: https://github.com/conda-forge/.github/blob/main/.github/ISSUE_TEMPLATE/1-bug-report.yml

GitHub... just renders it appropriately for editing and previewing

**Comment by milljm** [comment]
Is our syntax not triggering a GitHub style? What is the difference between:

https://github.com/idaholab/moose/blob/ec5de759e0f2ee370feb4db3527ce3d977540c47/.github/DISCUSSION_TEMPLATE/Q%26A%20General.yml

and

https://github.com/conda-forge/.github/blob/main/.github/ISSUE_TEMPLATE/1-bug-report.yml?plain=1

**Comment by GiudGiud** [comment]
> Is our syntax not triggering a GitHub style? What is the difference between:
> 
> https://github.com/idaholab/moose/blob/ec5de759e0f2ee370feb4db3527ce3d977540c47/.github/DISCUSSION_TEMPLATE/Q%26A%20General.yml
> 
> and
> 
> https://github.com/conda-forge/.github/blob/main/.github/ISSUE_TEMPLATE/1-bug-report.yml

Not really a difference. One is for issues one for discussions.
I m just following the template

File: .github/DISCUSSION_TEMPLATE/Q&A General.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        # Type your question here
+  - type: checkboxes
+    attributes:
+      label: Check these boxes if you have followed the posting rules.
+      options:
+        - label: Q&A General is the most appropriate section for my question
+          required: true
+        - label: I have consulted the posting Guidelines on the Discussions front page
+          required: true
+        - label: I have searched the forum and this question has not been asked before, including by myself
```
Comment (lindsayad) [inline .github/DISCUSSION_TEMPLATE/Q&A General.yml:15]:
```suggestion
        - label: I have searched the forum and this question has not been asked before
```

File: .github/DISCUSSION_TEMPLATE/Q&A General.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+          required: true
+        - label: I have search the MOOSE website and the documentation does not answer my question
```
Comment (lindsayad) [inline .github/DISCUSSION_TEMPLATE/Q&A General.yml:17]:
```suggestion
        - label: I have searched the MOOSE website and the documentation does not answer my question
```

File: .github/DISCUSSION_TEMPLATE/Q&A General.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+          required: true
+        - label: I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)
+          required: true
+        - label: I am not trying to get support on a year-old version of MOOSE
```
Comment (lindsayad) [inline .github/DISCUSSION_TEMPLATE/Q&A General.yml:21]:
```suggestion
        - label: I am not trying to get support on an old version of MOOSE
```

**Review by lindsayad — APPROVED** [review]

**Pull request merged**